Introduces XFF configuration support for Istio via CRD (#16103)
kyma-project/kyma: docs/05-technical-reference/00-custom-resources/oper-01-istio.md

---
title: Istio
type: Custom Resource
---

The `istios.operator.kyma-project.io` CustomResourceDefinition (CRD) is a detailed description of the kind of data and the format the Reconciler use to configure and install Istio. To get the up-to-date CRD and show the output in the `yaml` format, run this command:

```shell
kubectl get crd istios.operator.kyma-project.io -o yaml
```

## Sample custom resource

This is a sample Istio custom resource (CR) that the Reconciler will use to configure and install Istio. This example shows the single supported **numTrustedProxies** configuration setting. There must be only one Istio CR on the cluster.

```yaml
apiVersion: operator.kyma-project.io/v1alpha1
kind: Istio
metadata:
  name: istio
  labels:
    app.kubernetes.io/name: istio
spec:
  config:
    numTrustedProxies: 1
```

This table lists all the possible parameters of a given resource together with their descriptions:

| Field   |      Mandatory      |  Description |
|---|:---:|---|
| **metadata.name** | **YES** | Specifies the name of the Istio Manager CR. |

> | **spec.config.numTrustedProxies** | **NO** | Specifies the number of trusted proxies, [more info here](https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/). |

Reviewer: ```suggestion
| **spec.config.numTrustedProxies** | **NO** | Specifies the number of trusted proxies; for more info, read the [Istio documentation](https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/). |
```

> | **spec.config.numTrustedProxies** | **NO** | Specifies the number of trusted proxies; for more info, read the [Istio documentation](https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/). |

Reviewer: ```suggestion
| **spec.config.numTrustedProxies** | **NO** | Specifies the number of trusted proxies; for more information, read the [Istio documentation](https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/). |
```

> | **spec.config.numTrustedProxies** | **NO** | Specifies the number of trusted proxies; for more info, read the [Istio documentation](https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/). |

Reviewer: ```suggestion
| **spec.config.numTrustedProxies** | **NO** | Specifies the number of trusted proxies; for more information, read the [Istio documentation](https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/). |
```

---
title: Istio Manager Operator
---

The `istios.operator.kyma-project.io` CustomResourceDefinition (CRD) is a detailed description of the kind of data and the format the Istio Manager Operator listens for. To get the up-to-date CRD and show the output in the `yaml` format, run this command:

```shell
kubectl get crd istios.operator.kyma-project.io -o yaml
```

> More information on the Module Manager you may find [in here](https://github.com/kyma-project/module-manager/).

Reviewer 2: how's that information relevant here? what does it bring?

Author: Will remove it, not relevant for the customer.

> More information on the Module Manager you may find [in here](https://github.com/kyma-project/module-manager/).

Author: Will remove it, not relevant for the customer.

## Sample custom resource

> This is a sample custom resource (CR) that the Istio Manager Operator listens for to manage Istio installation on the cluster. This example has only the **numTrustedProxies** configuration setting which makes the Istio Manager Operator install and configure Istio appropriately. There must be only a single Istio Manager CR in the cluster.

Reviewer 2: @pbochynski yes ofc they are already included in design and documented in API proposal in Kyma Istio repository. What we are adding and documenting here is strictly about what's implemented: `numTrustedProxies`. should we link in this document to API proposal in Kyma Istio repo?

> This is a sample custom resource (CR) that the Istio Manager Operator listens for to manage Istio installation on the cluster. This example has only the **numTrustedProxies** configuration setting which makes the Istio Manager Operator install and configure Istio appropriately. There must be only a single Istio Manager CR in the cluster.

Reviewer 3: It is just a configuration you pass to istio operator resource. I expected it to be introduced in one shot. Do we need one sprint for each config option?

> This is a sample custom resource (CR) that the Istio Manager Operator listens for to manage Istio installation on the cluster. This example has only the **numTrustedProxies** configuration setting which makes the Istio Manager Operator install and configure Istio appropriately. There must be only a single Istio Manager CR in the cluster.

Reviewer 4: <img width="485" alt="image" src="https://user-images.githubusercontent.com/29116492/202188053-19708791-b936-4be7-b3a0-438eb6625320.png">

> This is a sample custom resource (CR) that the Istio Manager Operator listens for to manage Istio installation on the cluster. This example has only the **numTrustedProxies** configuration setting which makes the Istio Manager Operator install and configure Istio appropriately. There must be only a single Istio Manager CR in the cluster.

Reviewer 3: They are already implemented 😛 
It is only about making them available for end-user. @tomekpapiernik I see you are bored - should we expect some PRs also?

> This is a sample custom resource (CR) that the Istio Manager Operator listens for to manage Istio installation on the cluster. This example has only the **numTrustedProxies** configuration setting which makes the Istio Manager Operator install and configure Istio appropriately. There must be only a single Istio Manager CR in the cluster.

Author: This was clarified, for the moment we will only expose `numTrustedProxies` as configuration since this is the only field that we merge from the CR to the Istio configuration while installing/updating.
